integrations/website-indexer/src/index.ts

```diff
@@ -0,0 +1,277 @@
+import { z, RuntimeError } from '@botpress/sdk'
+import axios from 'axios'
+import _ from 'lodash'
+import parseRobots from 'robots-parser'
+import Url from 'url'
+// import xml2json from 'xml2json'
+import xml2js from 'xml2js'
+import { Scraper } from './scraper'
+import * as bp from '.botpress'
+
+export default new bp.Integration({
+  register: async () => {
+    /**
+     * This is called when a bot installs the integration.
+     * You should use this handler to instanciate ressources in the external service and ensure that the configuration is valid.
+     */
+    throw new RuntimeError('Invalid configuration') // replace this with your own validation logic
+  },
+  unregister: async () => {
+    /**
+     * This is called when a bot removes the integration.
+     * You should use this handler to instanciate ressources in the external service and ensure that the configuration is valid.
+     */
+    throw new RuntimeError('Invalid configuration') // replace this with your own validation logic
+  },
+  actions: {
+    indexUrls: async ({ input, logger, client }) => {
+      const pageUrlsSchema = z.object({
+        urls: z.array(z.string()),
+      })
+      const { urls: pageUrls } = pageUrlsSchema.parse(JSON.parse(input.pageUrls))
```

Comment:
:yellow_circle: **issue (non-blocking):** there should be a limit on the number of urls per request